src/cc/bcc_elf.c

```diff
@@ -646,6 +682,64 @@ static int find_debug_via_symfs(Elf *e, const char *path,
   return -1;
 }
 
+#ifdef HAVE_LIBLZMA
+
+#define LZMA_MIN_BUFFER_SIZE 4096
+static int open_mini_debug_info_file(void *gnu_debugdata,
+                                     size_t gnu_debugdata_size,
+                                     struct elf_file *out) {
+  void *decompressed = NULL;
+  size_t decompressed_data_size = 0;
+  size_t decompressed_buffer_size = 0;
+  lzma_stream stream = LZMA_STREAM_INIT;
+  lzma_ret ret;
+
+  ret = lzma_stream_decoder(&stream, UINT64_MAX, 0);
+  if (ret != LZMA_OK)
+    return -1;
+
+  stream.next_in = gnu_debugdata;
+  stream.avail_in = gnu_debugdata_size;
+  stream.avail_out = 0;
+
+  while (ret == LZMA_OK && stream.avail_in > 0) {
+    if (stream.avail_out < LZMA_MIN_BUFFER_SIZE) {
+      decompressed_buffer_size += LZMA_MIN_BUFFER_SIZE;
+      decompressed = realloc(decompressed, decompressed_buffer_size);
+      if (decompressed == NULL) {
```

Comment:
if `realloc` fails it will return `NULL` but won't free the buffer passed in in 1st arg, so you'll need to do some pointer juggling to free the original buffer here